Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -1581,6 +1646,142 @@ void ACesium3DTileset::showTilesToRender(
   }
 }
 
+namespace {
+
+template <typename Tag, typename Tag::type M> struct Rob {
+public:
+  friend typename Tag::type get(Tag) { return M; }
+};
+
+// Hack to retrieve the private ULocalPlayer::ViewStates
+class ULocalPlayer_Hack {
+public:
+  typedef TArray<FSceneViewStateReference> ULocalPlayer::*type;
+  friend type get(ULocalPlayer_Hack);
+};
+
+template struct Rob<ULocalPlayer_Hack, &ULocalPlayer::ViewStates>;
+} // namespace
+
+void ACesium3DTileset::RetrieveOcclusionResults(
+    TArray<FSceneViewState*>&& views) {
+  this->OcclusionResults.clear();
+  this->OcclusionResults.reserve(this->OcclusionRequests.size());
+  this->WaitingForOcclusion = true;
+  ENQUEUE_RENDER_COMMAND(RetrieveOccludedPrimitives)
+  ([views = std::move(views),
+    &WaitingForOcclusion = this->WaitingForOcclusion,
+    &OcclusionRequests = this->OcclusionRequests,
+    &OcclusionResults = this->OcclusionResults](FRHICommandList& RHICmdList) {
```

Comment:
I'm pretty sure this isn't thread safe. The ACesium3DTileset Actor can get garbage collected between the time the render command is created and the time it is executed, and if that happens the render thread will be writing to who-knows-where. The [Threaded Rendering docs](https://docs.unrealengine.com/4.27/en-US/ProgrammingAndScripting/Rendering/ThreadedRendering/) say as much:

> The game thread may delete a UObject while the rendering thread is processing a command that references it. For this reason, the rendering thread should never dereference a UObject pointer unless a mechanism is in place to make sure the UObject is not deleted until the rendering thread no longer references it. An example is UPrimitiveComponent, which uses a FRenderCommandFence called DetachFence to prevent GC from deleting the UObject before the rendering thread has processed the detach command.

Also, sort of unrelated: can/should we use `FRenderCommandFence` instead of the `WaitingForOcclusion` atomic?

Reply:
Resolving this one since we no longer need to do this on the render thread after using the ISceneViewExtension hook.